app/models/ca_sets.php

```diff
@@ -883,6 +883,373 @@ public function getSets($pa_options=null) {
 		}
 	}
 	# ------------------------------------------------------
+	# Set lists by SQL
+	# ------------------------------------------------------
+	/**
+	 * Replicates SetLists() but uses SQL to implement paging for performance reasons.
+	 *
+	 * @param array $pa_options Array of options. Supported options are:
+	 *			table - if set, list is restricted to sets that can contain the specified item. You can pass a table name or number. If omitted sets containing any content will be returned.
+	 *			setType - Restricts returned sets to those of the specified type. You can pass a type_id or list item code for the set type. If omitted sets are returned regardless of type.
+	 *			user_id - Restricts returned sets to those accessible by the current user. If omitted then all sets, regardless of access are returned.
+	 *			access - Restricts returned sets to those with at least the specified access level for the specified user. If user_id is omitted then this option has no effect. If user_id is set and this option is omitted, then sets where the user has at least read access will be returned.
+	 *			checkAccess - Restricts returned sets to those with an public access level with the specified values. If omitted sets are returned regardless of public access (ca_sets.access) value. Can be a single value or array if you wish to filter on multiple public access values.
+	 *			row_id = if set to an integer only sets containing the specified row are returned
+	 *			setIDsOnly = if set to true only set_id values are returned, in a simple array
+	 *			omitCounts =
+	 *			all =
+	 *			allUsers =
+	 *			publicUsers =
+	 *			name =
+	 *          byUser = return sets grouped by user with access. The array will be key'ed by sortable user name. Each entry includes a 'user' array with information about the user and a 'sets' array with the list of sets that user has access to. [Default is false]
+	 *          codes =
+	 *          includeItems =
+	 *          item_id = Get set that contains specified item_id
+	 *
+	 * @return array A list of sets keyed by set_id and then locale_id. Keys for the per-locale value array include: set_id, set_code, status, public access, owner user_id, content table_num, set type_id, set name, number of items in the set (item_count), set type name for display and set content type name for display. If setIDsOnly option is set then a simple array of set_id values is returned instead.
+	 */
+	public function getSetsBySQL($pa_options=null) {
```

Comment:
I know this is copypasta - I'm wondering whether it's possible to change the getSets to take an argument which makes it return the sql that this function can then use / modify to add the sort / limit parameters to. Perfectly fine for you to say - nah - too hard ;)

Reply:
@kehh The issue I see with doing it that way is that there are 3(?) separate queries that would need to be returned and run in a different order depending on which way we are doing it - counts go first currently, but I need to have the paged list of items to limit the query.

I would ideally re-write the whole thing, but we discussed leaving the current function as is so we don't need to retest other parts of the system that use the original function, so I was trying to mirror as closely as possible (thinking about future upgrades primarily)

Happy to go either way on this, but ideally I'd prefer to get this up somewhere we can test to ensure it actually has the desired impact first?